bot/code_review_bot/cli.py

```diff
@@ -162,6 +165,30 @@ def main():
             w.ingest_autoland(revision)
         else:
             w.run(revision)
+    except requests.exceptions.HTTPError as e:
+        logger.error("Static analysis http failure", revision=revision, error=e)
+
+        failure = UnitResult(
+            namespace="code-review",
+            name="general",
+            result=UnitResultState.Broken,
+            details="WARNING: An error occurred in the code review bot.\n\n```{}```".format(
+                str(e)
+            ),
+            format="remarkup",
+            duration=0,
+        )
+
+        w.phabricator.update_build_target(
+            revision.target_phid, BuildState.Fail, unit=[failure]
+        )
+
+        # Update Harbormaster status
+        w.update_status(revision, state=BuildState.Fail)
+
+        # Then raise to mark task as erroneous
+        raise
+
     except Exception as e:
```

Comment:
The problem in #984 is that when an exception happens from anywhere in the bot, we report a build error but we don't say anything to the user.
Instead of handling HTTP errors specifically, let's just report the generic failure whenever there is any exception.

Reply:
OK, I see now, this makes sense.